pgdrive/scene_creator/blocks/block.py

```diff
@@ -449,25 +466,53 @@ def _add_land(self, lanes):
                     theta = -numpy.arctan2(direction_v[1], direction_v[0])
                     width = lane.width_at(0) + self.SIDE_WALK_LINE_DIST * 2
                     length = lane.length
-                    self._add_land2bullet(middle, width, length * 1.3 / segment_num, theta)
-
-    def _add_land2bullet(self, middle, width, length, theta):
-        cm = CardMaker('card')
-        cm.setFrame(-length / 2, length / 2, -width / 2, width / 2)
-        cm.setHasNormals(True)
-        cm.setUvRange((0, 0), (length / 20, width / 10))
-        card = self.card_node_path.attachNewNode(cm.generate())
-        card.setPos(middle[0], -middle[1], numpy.random.rand() * 0.01 - 0.01)
+                    self._add_lane2bullet(middle, width, length * 1.3 / segment_num, theta, lane, (from_, to_, index))
 
-        card.setQuat(
+    def _add_lane2bullet(self, middle, width, length, theta, lane: Union[StraightLane, CircularLane], lane_index):
+        """
+        Add lane visualization and body for it
+        :param middle: Middle point
+        :param width: Lane width
+        :param length: Segment length
+        :param theta: Rotate theta
+        :param lane: Lane info
+        :return: None
+        """
+        segment_np = NodePath(LaneNode(BodyName.Lane, lane, lane_index))
+        segment_node = segment_np.node()
+        segment_node.setActive(False)
+        segment_node.setKinematic(False)
+        segment_node.setStatic(True)
+        shape = BulletBoxShape(Vec3(length / 2, 0.1, width / 2))
+        segment_node.addShape(shape)
+        segment_node.setIntoCollideMask(BitMask32.bit(Block.LANE_SURFACE_COLLISION_MASK))
+        self.static_nodes.append(segment_node)
+        segment_np.setPos(middle[0], -middle[1], -0.1)
+        segment_np.setQuat(
             LQuaternionf(
                 numpy.cos(theta / 2) * numpy.cos(-numpy.pi / 4),
                 numpy.cos(theta / 2) * numpy.sin(-numpy.pi / 4), -numpy.sin(theta / 2) * numpy.cos(-numpy.pi / 4),
                 numpy.sin(theta / 2) * numpy.cos(-numpy.pi / 4)
             )
         )
-        card.setTransparency(TransparencyAttrib.MMultisample)
-        card.setTexture(self.ts_color, self.road_texture)
+        segment_np.reparentTo(self.lane_node_path)
+        if self.render:
+            cm = CardMaker('card')
+            cm.setFrame(-length / 2, length / 2, -width / 2, width / 2)
+            cm.setHasNormals(True)
+            cm.setUvRange((0, 0), (length / 20, width / 10))
+            card = self.lane_vis_node_path.attachNewNode(cm.generate())
+            card.setPos(middle[0], -middle[1], numpy.random.rand() * 0.01 - 0.01)
```

Comment:
What this randomization means here? Should we use a random state to generate it to guarantee the reproducibility?

Reply:
No. it only affects the visualization instead of the physics world. The texture of the lane surface should be in different height. Otherwise, the lane surface will **flicker** when rendering.